ts/src/base/Exchange.ts

```diff
@@ -4202,7 +4202,11 @@ export default class Exchange {
             }
             return markets[0];
         }
-        throw new BadSymbol (this.id + ' does not have market symbol ' + symbol);
+        let symbolForLog = symbol;
+        if (typeof symbolForLog !== 'string') {
+            symbolForLog = this.json (symbolForLog);
+        }
+        throw new BadSymbol (this.id + ' does not have market symbol ' + symbolForLog);
```

Comment:
this is not an expected behavior, market never expects to get anything other than a string, we even have that type in the signature.